docs: Some cleanup, some v2.4.0 additions (#139)
caddyserver/website: src/docs/markdown/caddyfile/directives/handle.md

The `handle` directive is kind of similar to the `location` directive from nginx config: the first matching `handle` block will be evaluated. Handle blocks can be nested if needed. Only HTTP handler directives can be used inside handle blocks.

> <aside class="tip">

Reviewer: This is fine, but what about something like "Similar directives"? https://caddyserver.com/docs/caddyfile/directives/rewrite#similar-directives

If you prefer crafting HTTP handler logic in a more inheritence-based way like nginx location blocks, you may prefer the use of `handle` blocks rather than defining mutually-exclusive matchers for your directives. If inheritence is a desired characteristic of your HTTP handler configurations, then the `handle` directive may suit you well.

## Similar directives

> There are other directives that can wrap other HTTP handler directives, but but imply a different intent or behave subtly differently:

Reviewer: This section could use improvement, including in the other linked pages where this section exists. I'll just comment on this one.

```suggestion
There are other directives that can wrap HTTP handler directives, but each has its use depending on the behavior you want to convey:
```

Author: :+1: I was somewhat copying the wording from the `rewrite` page.

> - [`handle_path`](handle_path) does the same as `handle`, but also has built-in path prefix stripping logic built-in before handling the directives within.

Reviewer: ```suggestion
- [`handle_path`](handle_path) does the same as `handle`, but it strips a prefix from the request before running its handlers.
```

Just trying to make it a little easier to understand/read.

> - [`handle_errors`](handle_errors), similarly to `handle`, wraps other directives, but is instead invoked when Caddy encounters an error during request handling.

Reviewer: ```suggestion
- [`handle_errors`](handle_errors) is like `handle`, but is only invoiked when Caddy encounters an error during request handling.
```

> - [`route`](route) similarly to `handle`, wraps other directives, but instead of providing mutual exclusivity, allows overriding the default [directive order](/docs/caddyfile/directives#directive-order).

Reviewer: ```suggestion
- [`route`](route) wraps other directives like `handle` does, but with two distinctions: 1) route blocks are not mutually exclusive to each other, and 2) directives within a route are not [re-ordered]([directive order](/docs/caddyfile/directives#directive-order), giving you more control if needed.
```

	reverse_proxy 127.0.0.1:8080
}
```

> You can mix `handle` and [`handle_path`](handle_path) directives in the same site, and they will still be mutually exclusive from eachother:

Reviewer: ```suggestion
You can mix `handle` and [`handle_path`](handle_path) directives in the same site, and they will still be mutually exclusive from each other:
```
